lib/kb_trimmomatic/kb_trimmomaticImpl.py

```diff
@@ -766,20 +793,28 @@ def execTrimmomatic(self, ctx, input_params):
                                   #'info':
                                       })
             if some_trimmed_output_created:
+                single_reads = False
                 if read_type == 'SE':
                     reads_desc_ext = " Trimmomatic trimmed SingleEndLibrary"
                     reads_name_ext = "_trimm"
+                    single_reads = True
                 else:
                     reads_desc_ext = " Trimmomatic trimmed paired reads"
                     reads_name_ext = "_trimm_paired"
-                output_readsSet_obj = { 'description': input_readsSet_obj['data']['description']+reads_desc_ext,
+                output_readsSet_obj = { 'description': str(input_readsSet_obj['data']['description'])+reads_desc_ext,
                                         'items': items
                                         }
                 output_readsSet_name = str(input_params['output_reads_name'])+reads_name_ext
                 trimmed_readsSet_ref = setAPI_Client.save_reads_set_v1 ({'workspace_name': input_params['output_ws'],
                                                                          'output_object_name': output_readsSet_name,
                                                                          'data': output_readsSet_obj
                                                                          })['set_ref']
+                trimmed_RNASeqSampleSet_ref = self._save_RNASeqSampleSet(
```

Comment:
This should only get called if the input object was a SampleSet

Reply:
yea. it's in the if-else block. Only input is a set, it will get called.